src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -125,25 +125,30 @@
 
 (defn maybe-tweak-catalog
   "Slightly tweak the given catalog, returning a new catalog, `rand-percentage`
-  percent of the time."
-  [catalog rand-percentage]
-  (if (< (rand 100) rand-percentage)
-    (rand-catalog-mutation catalog)
-    catalog))
-
-(defn update-catalog [catalog]
-  (assoc catalog "producer_timestamp" (time/now)))
+   percent of the time."
+  [catalog rand-percentage stamp]
+  (let [catalog' (assoc catalog "producer_timestamp" stamp)]
+    (if (< (rand 100) rand-percentage)
+      (rand-catalog-mutation catalog')
+      catalog')))
+
+(defn jitter
+  "jitter a timestamp (rand-int n) seconds in the forward direction"
+  [stamp n]
+  (time/plus stamp (time/seconds (rand-int n))))
 
 (defn update-report-run-fields
   "configuration_version, start_time and end_time should always change
    on subsequent report submittions, this changes those fields to avoid
    computing the same hash again (causing constraint errors in the DB)"
-  [report]
-  (assoc report
-         "configuration_version" (ks/uuid)
-         "start_time" (time/now)
-         "end_time" (time/now)
-         "producer_timestamp" (time/now)))
+  [report stamp]
+  (-> report
+      (update "resource_events" (partial map #(assoc % "timestamp"
+                                                     (jitter stamp 300))))
```

Comment:
I'm not inclined to change it without a performance reason, just because I find specter's update uglier than update in core. Looking at a profiler, this doesn't seem like a particular hotspot in the benchmark code. Any reason to prefer specter here?